### DISCUSSION TITLE
files are written as user `root` (problematic for external libraries)

### DISCUSSION THREAD

**Discussion by C-Otto** [immich-app/immich#11825] [Q&A] [answered]
### The bug

Files written by immich always belong to the `root` user, which seems to be related to how Docker works. This isn't an issue for uploads and other files solely managed by immich itself, but it is an issue for external libraries, as these files are also read by other programs.

In my case, with an external library full of JPGs:

When immich creates an XMP file, for example when I update an asset's description or change the date, it creates a new XMP file. For `IMG_123.jpg`, a file `IMG_123.jpg.xmp` is created at the same location. While `IMG_123.jpg` might belong to my regular user (say, user id 1000), the XMP file belongs to `root`. As such, the file cannot be read by my regular user (and management tools like `git annex` which I run with that user).

I'd prefer immich to create files with a configurable user, possibly limited to external libraries. It might also be a good idea to re-use the user (and group) settings of the original file (`IMG_123.jpg` in my example).

### The OS that Immich Server is running on

Debian Stable

### Version of Immich Server

v1.112.0

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Reproduction steps

1. Add external library (RW)
2. Update metadata for asset in external library
3. Observe file details for created XMP file

**Comment by jrasm91** [comment, marked as answer]
Immich runs as root by default, there is a separate issue for that "bug". You can configure it to run as another user though.

https://github.com/immich-app/immich/issues/2750

https://immich.app/docs/FAQ#how-can-i-run-immich-as-a-non-root-user

**Reply by C-Otto** [reply]
Thanks, adding the user to my docker compose fixed the issue for me!

**Reply by mertalev** [reply]
I thought it ran as `node`? Huh, I guess not.
